### PR TITLE
Improvements to application management service layer

### PR DESCRIPTION
**Pull request by mefarazath**
* Expose STS admin operation to remove a trusted service through the interface. The implementation was already available as a public method but the interface did not expose it.

* Expose skipConsent and JWKS URI through SP model object.
Previously, the only way to get the values for the above was to read service provider properties and check the values. This is inconvenient since in the future we might change the way we store the property value.

This change abstracts how the properties are stored and exposes them directly through methods in the SP object.

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 1]
```diff
@@ -1721,94 +1723,22 @@ private void updatePermissionAndRoleConfiguration(int applicationID,
     }
 
     @Override
-    public ServiceProvider getApplication(String applicationName, String tenantDomain)
-            throws IdentityApplicationManagementException {
+    public ServiceProvider getApplication(String applicationName,
```
Comment (mefarazath) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Refactored this method to remove code duplication. We already have a method that loads an SP by internal id.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java [hunk 2]
```diff
@@ -490,6 +490,13 @@ public boolean isUseUserstoreDomainInLocalSubjectIdentifier() {
         return false;
     }
 
+    public boolean isSkipConsent() {
+        if (serviceProvider.getLocalAndOutBoundAuthenticationConfig() != null) {
```
Comment (vihanga-liyanage) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java]: New line?

Also, can we add a method comment pls?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 3]
```diff
@@ -1820,6 +1820,10 @@ private ServiceProvider doAddApplication(ServiceProvider serviceProvider, String
                 PermissionsAndRoleConfig permissionAndRoleConfig = serviceProvider.getPermissionAndRoleConfig();
                 ApplicationMgtUtil.storePermissions(applicationName, username, permissionAndRoleConfig);
             } catch (IdentityApplicationManagementException ex) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Creating application: " + applicationName + " in tenantDomain: " + tenantDomain +
+                            " failed. Rolling back by cleaning up partially deleting partial created data.");
```
Comment (vihanga-liyanage) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Need to fix the log.

Comment (vihanga-liyanage) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Need to fix the log.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java [hunk 4]
```diff
@@ -1080,48 +1087,8 @@ public void update(HttpServletRequest request) {
...
-            propertyForSkipConsent.setDisplayName(IdentityConstants.SKIP_CONSENT_DISPLAY_NAME);
-            propertyForSkipConsent.setName(IdentityConstants.SKIP_CONSENT);
-            propertyForSkipConsent.setValue(Boolean.toString(skipConsent));
-
-            configNameValueMap.put(propertyForSkipConsent.getName(), propertyForSkipConsent);
-        }
-
-        serviceProvider.setSpProperties(configNameValueMap.values().toArray(new ServiceProviderProperty[]{}));
+        String jwks = request.getParameter("jwksUri");
+        serviceProvider.setJwksUri(jwks);
```
Comment (malithie) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java]: Shall we use the constant defined

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 5]
```diff
@@ -1721,94 +1721,22 @@ private void updatePermissionAndRoleConfiguration(int applicationID,
...
-        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
         try {
-
-            // Load basic application data
-            ServiceProvider serviceProvider = getBasicApplicationData(applicationName, connection, tenantID);
-
-            if ((serviceProvider == null || serviceProvider.getApplicationName() == null)
-                    && ApplicationConstants.LOCAL_SP.equals(applicationName)) {
+            int applicationId = getApplicationIdByName(applicationName, tenantDomain);
+            if (applicationId == 0 && ApplicationConstants.LOCAL_SP.equals(applicationName)) {
```
Comment (malithie) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Why it's important to check for applicationId == 0

**Pull request merged**